devices/views.py

```diff
@@ -72,6 +72,55 @@
 from users.models import Lageruser
 
 
+def get_devices(user, category, department, sorting):
+    if category == "all":
+        devices = Device.objects.all()
+    elif category == "available":
+        devices = Device.active().filter(currentlending=None)
+    elif category == "lent":
+        lendings = Lending.objects.filter(returndate=None)
+        devices = Device.objects.filter(lending__in=lendings)
+    elif category == "archived":
+        devices = Device.objects.exclude(archived=None)
+    elif category == "trashed":
+        devices = Device.objects.exclude(trashed=None)
+    elif category == "overdue":
+        lendings = Lending.objects.filter(returndate=None, duedate__lt=datetime.date.today())
+        devices = Device.objects.filter(lending__in=lendings)
+    elif category == "returnsoon":
+        soon = datetime.date.today() + datetime.timedelta(days=10)
+        lendings = Lending.objects.filter(returndate=None, duedate__lte=soon,
+                                          duedate__gt=datetime.date.today())
+        devices = Device.objects.filter(lending__in=lendings)
+    elif category == "temporary":
+        devices = Device.active().filter(templending=True)
+    elif category == "bookmark" and user.is_authenticated:
+            devices = user.bookmarks.all()
+    elif category == "active":
+        devices = Device.active()
+    else:
+        raise ValueError
+
+    if department == "all":
+        pass
+    elif department == "my":
+        devices = devices.filter(department__in=user.departments.all())
+    elif department.isdigit():
+        devices = devices.filter(department_id=int(department))
+    else:
+        raise ValueError
+
+    if sorting not in [s[0] for s in VIEWSORTING_DEVICES]:
+        raise ValueError
+
+    if hasattr(user, 'departments'):
+        devices = devices.exclude(~Q(department__in=user.departments.all()), is_private=True)
+    else:
+        devices = devices.exclude(is_private=True)
```

Comment:
This `else` case did not exist before.

If a user has the `departments` property (probably the same as being authenticated) private devices from other departments were excluded already. In other words: I can only see private devices from my own department.

I think it makes sense to hide all private devices from users who do not even have a department.